functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/customAssertions/BuildProjectAssert.java

```diff
@@ -0,0 +1,59 @@
+package org.arquillian.smart.testing.ftest.customAssertions;
+
+import java.io.File;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.FileAssert;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.BuiltProject;
+
+public class BuildProjectAssert extends AbstractAssert<BuildProjectAssert, BuiltProject> {
+
+    BuildProjectAssert(BuiltProject actual) {
+        super(actual, BuildProjectAssert.class);
+    }
+
+    public static BuildProjectAssert assertThat(BuiltProject module) {
+        return new BuildProjectAssert(module);
+    }
+
+    public BuildProjectAssert hasAllBuiltSubModulesContainEffectivePom(String effectivePom) {
+        actual.getModules().forEach(subModule -> assertThat(subModule).containsEffectivePom(effectivePom));
+        return this;
+    }
+
+    public BuildProjectAssert hasAllBuiltSubModulesContainReport(String report) {
+        actual.getModules().forEach(subModule -> assertThat(subModule).hasReportFile(report));
+        return this;
+    }
+
+    private BuildProjectAssert containsEffectivePom(String effectivePom) {
+        if (actual.getTargetDirectory().exists()) {
+            FileAssert fileAssert = new FileAssert(new File(actual.getTargetDirectory(), effectivePom));
+            fileAssert.exists();
```

Comment:
The only thing that you assert here is that the file exists, nothing about the content or that it is really an effective pom and the correct one - generated from the pom in the related module